app/models/eligibilities/verification_history.rb

```diff
@@ -17,6 +17,14 @@ class VerificationHistory
     field :verification_outstanding, type: Boolean
     field :due_on, type: Date
     field :aasm_state, type: String
+    field :date_of_action, type: DateTime
 
+    after_create :set_date_of_action
+
+    private
+
+    def set_date_of_action
+      write_attribute(:date_of_action, DateTime.now)
```

Comment:
@jacobkagon Can you add a check to see if the value is already populated here? And only update if the value is blank.